### DISCUSSION TITLE
Can not move bone or a node

### DISCUSSION THREAD

**Discussion by xyz99002** [stride3d/stride#2390] [Newcomers] [answered]
Hi, I am not a gaming person but trying to learn 3D for industrial application. I am trying move a bone or a node of my model. I have posted the code below and expecting it should move (change the position) the bone or node but it does not. The similar code does move it in Unity. The stride model are reprented as model and skeleton different then unity and that is where I guess I am having issue.The thing I am trying to do should be very simple however I am not sure how to do it properly in Stride. I would appreciate any help on this. I expect to move M2_M1 node(with child) to move by 5 units.

        public Entity robot;

        private Vector3 zPosition;
        private Vector3 sPosition;

        TransformTRS zTransform;

        public override void Start()
        {
            var test = robot.Get<ModelComponent>().Skeleton;

            for (int i = 0; i < test.Nodes.Length; i++)
            {
                if (test.Nodes[i].Name == "M2_M1")
                {
                    zPosition = test.Nodes[i].Transform.Position;
                    zTransform = test.Nodes[i].Transform;
                }

            }
        }

        public override void Update()
        {
            zTransform.Position = zPosition + new Vector3(0, 5, 0);
        }

**Comment by DJDQfff** [comment]
zPosition is  a struct, not object

**Reply by xyz99002** [reply]
It is used to set the new position using node transform in Update(). So, what is wrong with that?

**Comment by IXLLEGACYIXL** [comment]
you have the bone link component 
which is bound to a bone in your skeleton, try using that one

**Reply by IXLLEGACYIXL** [reply]
thats just the alias in editor to use in code use ModelNodeLinkComponent

**Reply by xyz99002** [reply]
Thank you for the suggestion. I added a line in the code to get bone link "robot.Get\<ModelNodeLinkComponent\>();". But this returns null. Am I missing anything?

**Reply by IXLLEGACYIXL** [reply]
yes you need to attach a bone link in your editor and link it to a bone in your sceleton
also get component is not recommended

use `public ModelNodeLinkComponent BoneLink {get; init;}` then oyu can set it through editor

then you realize you dont have something to set as you are missing the bone link component in your editor

**Reply by xyz99002** [reply]
You mean add bone link to my model and assign the skeleton nodes to the links? Also BoneLinks do not have transforms so how can they be moved?

**Comment by xyz99002** [comment, marked as answer]
I have my code working now after changing like as below. It was not working because Transform is a structure and hence copying it to zTransform was copying the data and updating was updatig the copy. In Unity it is a class so it works. To make it work in Stride, I keep reference to the array which is assigned by reference. Directly changing the stricture using that reference resulted in the node/bone moving. 

Thank you for your support.

    public Entity robot;

    private Vector3 zPosition;

    ModelNodeTransformation [] transforms;
    int zTransformIndex;

    public override void Start()
    {
        var test = robot.Get<ModelComponent>().Skeleton;

        for (int i = 0; i < test.Nodes.Length; i++)
        {
            if (test.Nodes[i].Name == "M2_M1")
            {
                zPosition = test.Nodes[i].Transform.Position;
                transforms = test.NodeTransformations;
                zTransformIndex = i;
            }

        }
    }

    public override void Update()
    {
        transforms[zTransformIndex].Transform.Position = zPosition + new Vector3(0, 5, 0);
    }
